racket/collects/version/check.rkt

```diff
@@ -93,6 +104,11 @@
 (define (check-version)
   (with-timeout timeout check-version-raw))
 
+(define simulate-timeout-for-interactive-testing? #f)
```

Comment:
Might be nice to read this from an environment variable.

Reply:
Yes, I'll do that.

Reply:
How about making it a parameter that is only set by the testing submodule?

Reply:
I think it's also an issue for clients of this module: I would not have expected calling `check-version` to potentially block for 30 seconds!

Reply:
I'm thinking now about adding a `check-version-evt` and/or making the timeout an optional argument, but I also don't want to make this code more complex, especially since it doesn't seem to have a lot of eyeballs on it.

Reply:
That seems like overkill, I think either my idea or @rmculpepper's is fine and you should just pick whichever seems better.